nupic/research/frameworks/vernon/mixins/oml.py

```diff
@@ -123,21 +155,53 @@ def create_test_test_dataloader(cls, config, dataset):
             pin_memory=torch.cuda.is_available(),
         )
 
+    def pre_task(self, tasks):
+        """Re-initialize task params prior to meta-train training."""
+        super().pre_task(tasks)
+        if not self.reset_task_params:
+            return
+
+        output_params = self.get_named_output_params().values()
+        for p in output_params:
+            if p.dim() == 2:
+                for t in tasks:
+                    task_weights = p[t, :].unsqueeze(0)
+                    nn.init.kaiming_normal_(task_weights)
+
     def post_epoch(self):
         super().post_epoch()
-        if self.should_stop() and self.run_meta_test:
-            # meta-training phase complete, perform meta-testing phase
-            for num_classes_learned in [10, 50, 100, 200, 600]:
-                if num_classes_learned > self.num_classes_eval:
+        if self.current_epoch >= self.epochs - 1 and self.run_meta_test:
+
+            # Accumulate results.
+            table = []
+            headers = ["Num Classes", "Meta-test test", "Meta-test train", "LR"]
+
+            # Meta-training phase complete, perform meta-testing phase
+            for num_classes in self.num_meta_test_classes:
+                if num_classes > self.num_classes_eval:
```

Comment:
I didn't get the meaning behind this if statement, a comment would help

Reply:
The `self.num_classes_eval` are the number of meta-testing classes. This prevents sampling, say, 700 classes from a meta-test set comprised of only 659 classes.